### ISSUE TITLE
Add `count` prop to OuiTreeView

### ISSUE THREAD

**Issue by KrooshalUX** [opensearch-project/oui#934]
- [ ] `Count` prop should reflect the number of `OuiTreeNode` children, and should be optional to include

<img width="351" alt="Screen Shot 2023-07-28 at 12 35 57 AM" src="https://github.com/opensearch-project/oui/assets/105819041/1533f0e2-e177-4f64-a844-60286c9824fc">

 
- [ ] Reference `count` styling from `OuiFacet` and `OuiFilterGroup` to build styling while retaining hover styling of `OuiTreeView`  - spacing between text & count = `$ouiSizeS`
**Examples:**
**- Default:** 
<img width="185" alt="Screen Shot 2023-07-28 at 12 04 09 AM" src="https://github.com/opensearch-project/oui/assets/105819041/54a32dc4-9eb3-485a-8ecb-80878cd973d9">

<img width="173" alt="Screen Shot 2023-07-28 at 12 08 03 AM" src="https://github.com/opensearch-project/oui/assets/105819041/747166f5-aff2-4145-ae56-cc674fd83401">

**- Disabled:** 
<img width="134" alt="Screen Shot 2023-07-28 at 12 04 14 AM" src="https://github.com/opensearch-project/oui/assets/105819041/5a7dd6cc-4220-4027-80b7-a7b2247c2b5d">

Discuss:
What do we think about using the styling of `selected` that is present in `OuiFacet` and `OuiFilterGroup` for the `isExpanded` state of `OuiTreeGroup` ? Do we think that the expansion itself is visual communication or not? IMO, the expansion itself is sufficient, but for the sake of complete exploration here are examples of count changing color when a user is interacting with an element that contains a count: 

<img width="175" alt="Screen Shot 2023-07-28 at 12 08 55 AM" src="https://github.com/opensearch-project/oui/assets/105819041/c3c9792e-a9c8-4cec-954e-0a81ed232faa">
<img width="165" alt="Screen Shot 2023-07-28 at 12 09 03 AM" src="https://github.com/opensearch-project/oui/assets/105819041/ca0bcce7-21d8-4a4d-bd5f-c81c416d3314">

**Comment by pjfitzgibbons** [comment]
@anirudha

**Comment by KrooshalUX** [comment]
@pjfitzgibbons I may have caused some confusion by referencing facet styling here. OuiTreeView already exists and the expand / collapse icons can be set by the builder - they can be chevrons or other icons (like folders) : https://oui.opensearch.org/1.2/#/navigation/tree-view

Since count would be optional in OuiTreeNav, we would still want to use expand/collapse indicators. Builders may also choose to force expand the tree as well.

For clarity - this component doesn't contain facet - for the sake of speed/ not having to red-line a design or cause an inconsistency, I was pointing to the design of count on OuiFacet and OuiFilterGroup for the builder to reference. 

Can you take a look at the OUI Docs for Tree view and see if the way icons for expand/collapse give you any concerns or questions ?

**Comment by pjfitzgibbons** [comment]
OK. looking.  You references "OuiTreeNav".. are you considering adding a new/fork component here?   I don't understand the difference vs OuiTreeView.
And I didn't realize OuiTreeView was pre-existing... will investigate.

**Comment by KrooshalUX** [comment]
@pjfitzgibbons No, not a fork or a new component, this is just an additional prop within the existing component. Right now there is no count in tree view today. 

Here is what tree view looks like currently: 

<img width="1045" alt="Screen Shot 2023-07-28 at 11 05 18 AM" src="https://github.com/opensearch-project/oui/assets/105819041/05809c95-afa7-47ee-b3ff-d9a2ada64196">
<img width="291" alt="Screen Shot 2023-07-28 at 11 05 26 AM" src="https://github.com/opensearch-project/oui/assets/105819041/4cb867b6-21de-46c0-b7e7-f27cdee0d35e">

**Comment by keskami** [comment]
I can take this issue. Basically a count prop that is displayed next to the labels of the tree view that mirrors OuiFacet's count prop?

**Comment by anirudha** [comment]
Closing — this issue has been open for 1-3 years without progress. If still relevant, please reopen or create a new issue with updated context.
